src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_IncrementalAnalyzer.cs

```diff
@@ -75,30 +75,6 @@ private async Task TextDocumentOpenAsync(TextDocument document, CancellationToke
         /// </summary>
         private ImmutableArray<StateSet> GetStateSetsForFullSolutionAnalysis(ImmutableArray<StateSet> stateSets, Project project)
         {
-            // If full analysis is off, remove state that is created from build.
-            // this will make sure diagnostics from build (converted from build to live) will never be cleared
-            // until next build.
-            _ = GlobalOptions.IsFullSolutionAnalysisEnabled(project.Language, out var compilerFullSolutionAnalysisEnabled, out var analyzersFullSolutionAnalysisEnabled);
-            if (!compilerFullSolutionAnalysisEnabled)
-            {
-                // Full solution analysis is not enabled for compiler diagnostics,
-                // so we remove the compiler analyzer state sets that are from build.
-                // We do so by retaining only those state sets that are
-                // either not for compiler analyzer or those which are for compiler
-                // analyzer, but not from build.
-                stateSets = stateSets.WhereAsArray(s => !s.Analyzer.IsCompilerAnalyzer() || !s.FromBuild(project.Id));
```

Comment:
`!s.FromBuild(project.Id)` is now always true.  So this is a WhereAsArray(true) which is a no-op.  so entirely removed.